### PR TITLE
Pickups

### PR DESCRIPTION
**Pull request by shashagit**
Temporary pickups spawn when number of enemies killed is in a random range of 5 to 9. Most of them are temporary buffs.
Permanent pickups use the same prefabs as temp but have a permanent effect time. The buff value is 1.2-2.0x more than temp buffs randomly.

### CODE REVIEW

File: HollowEngine/BulletHell/src/GameLogic/GameLogicManager.cpp [hunk 1]
```diff
@@ -195,6 +184,54 @@ namespace BulletHell
...
+		{
+			pStats->mFireRate += pPickup->mBuffValue;
+			break;
+		}
+		case PickupType::INVINCIBILITY:
+		{
+			Health* pHp = pPlayer->GetComponent<Health>();
+			pHp->mInvincibleTime += pPickup->mBuffValue;
+			break;
+			break;
```
Comment (drasske) [inline HollowEngine/BulletHell/src/GameLogic/GameLogicManager.cpp]: Double break hype

File: HollowEngine/BulletHell/src/Events/PickupTimedEvent.h [hunk 2]
```diff
@@ -13,9 +13,9 @@ namespace BulletHell
 	class PickupTimedEvent : public Hollow::GameEvent
 	{
 	public:
-		PickupTimedEvent(Pickup* p) : Hollow::GameEvent((int)GameEventType::ON_PICKUP_COLLECT), mpPickup(p) {}
+		PickupTimedEvent(Pickup* p) : Hollow::GameEvent((int)GameEventType::ON_PICKUP_EFFECT_END), mpPickup(*p) {}
 	public:
-		Pickup* mpPickup;
+		Pickup mpPickup;
```
Comment (theAlekseyP) [inline HollowEngine/BulletHell/src/Events/PickupTimedEvent.h]: mPickup instead of mpPickaup

Comment (shashagit) [reply HollowEngine/BulletHell/src/Events/PickupTimedEvent.h]: good catch. Fixed.

File: HollowEngine/BulletHell/src/GameLogic/GameLogicManager.cpp (hunk 1 above)
Comment (shashagit) [reply HollowEngine/BulletHell/src/GameLogic/GameLogicManager.cpp]: zeleeted.

**Pull request merged**